substrate/bits

```diff
@@ -51,12 +51,16 @@ namespace substrate
 		);
 	}
 
-	/* TODO: Determin if we can drop this and abstract rotl/rotr to shift by the fixed nybble amount */
+	/* TODO: Determine if we can drop this and abstract rotl/rotr to shift by the fixed nibble amount */
 	template<typename T> SUBSTRATE_NO_DISCARD(typename std::enable_if<std::is_integral<T>::value ,T>::type
-		constexpr inline shift_nibble(const T v, const std::size_t shift) noexcept)
+		constexpr inline shift_nibble(const T value, const std::size_t shift) noexcept)
```

Comment:
This might need to be SUBSTRATE_CXX14_CONSTEXPR now the function has more than a single return statement as the body